src/dashboard/src/settings/components/oidc_auth.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+# AUTH_SERVER = 'https://login.microsoftonline.com/common/v2.0/'
+OIDC_RP_CLIENT_ID = os.environ.get("OIDC_RP_CLIENT_ID", "")
+OIDC_RP_CLIENT_SECRET = os.environ.get("OIDC_RP_CLIENT_SECRET", "")
+
+OIDC_OP_AUTHORIZATION_ENDPOINT = ""
+OIDC_OP_TOKEN_ENDPOINT = ""
+OIDC_OP_USER_ENDPOINT = ""
+OIDC_OP_JWKS_ENDPOINT = ""
+
+AZURE_TENANT_ID = os.environ.get("AZURE_TENANT_ID", "")
+if AZURE_TENANT_ID:
+    OIDC_OP_AUTHORIZATION_ENDPOINT = (
+        "https://login.microsoftonline.com/%s/oauth2/v2.0/authorize" % AZURE_TENANT_ID
+    )
+    OIDC_OP_TOKEN_ENDPOINT = (
+        "https://login.microsoftonline.com/%s/oauth2/v2.0/token" % AZURE_TENANT_ID
+    )
+    OIDC_OP_USER_ENDPOINT = (
+        "https://login.microsoftonline.com/%s/openid/userinfo" % AZURE_TENANT_ID
+    )
+    OIDC_OP_JWKS_ENDPOINT = (
+        "https://login.microsoftonline.com/%s/discovery/v2.0/keys" % AZURE_TENANT_ID
+    )
+else:
+    OIDC_OP_AUTHORIZATION_ENDPOINT = os.environ.get("OIDC_OP_AUTHORIZATION_ENDPOINT", "")
+    OIDC_OP_TOKEN_ENDPOINT = os.environ.get("OIDC_OP_TOKEN_ENDPOINT", "")
+    OIDC_OP_USER_ENDPOINT = os.environ.get("OIDC_OP_USER_ENDPOINT", "")
+    OIDC_OP_JWKS_ENDPOINT = os.environ.get("OIDC_OP_JWKS_ENDPOINT", "")
```

Comment:
These settings: is there any scenario in which you'd want to set just some of them, but not all of them?

I wonder if we should this, so it asks for all or none, and errors if you supply an incomplete set. Something like:

```python
if is_using_oidc():
    OIDC_OP_AUTH_ENDPOINT = os.environ["OIDC_OP_AUTH_ENDPOINT"]
    ...
else:
    OIDC_OP_AUTH_ENDPOINT = ""
    ...
```

So if you supply only half the keys, you get a big red KeyError in your startup logs.

Reply:
Yes - I think the oidc lib itself will complain (although I'm not sure it'll flag them up for being empty rather than not present) - I'll check this.